applications/openshift/api-server/api_server_tls_security_profile_custom_min_tls_version/rule.yml

```diff
@@ -0,0 +1,61 @@
+documentation_complete: true
+
+prodtype: ocp4
+
+title: "Check if API server is configured to use Old tlsSecurityProfile"
+
+description: |-
+    The configuration <tt>tlsSecurityProfile</tt> specifies TLS configurations
+    to be used while establishing connections with the externally exposed
+    servers. Though secure transport mode is used for establishing connections,
+    the protocols used may not always be strong enough to avoid interception and
+    manipulation of the data in transport. When Custom TLS Security profile is 
+    used it's always better to configure TLS version 1.2 or newer to avoid any
+    security breaches.
+
+    Update minTLSVersion configured in Custom tlsSecurityProfile using the following command:
+    {{% raw %}}<pre>oc patch apiservers.config.openshift.io cluster --type 'merge' --patch '{"spec":{"tlsSecurityProfile":{"custom":{"minTLSVersion":"VersionTLS12"}}}}'</pre>{{% endraw %}}
+
+    For more information, follow
+    {{{ weblink(link="https://docs.openshift.com/container-platform/latest/security/tls-security-profiles.html",
+                text="the relevant documentation") }}}.
+
+rationale: |-
+    The authenticity and integrity of the container platform and communication
+    between nodes and components must be secure. If an insecure protocol,
+    cipher, or algorithms is used, during transmission of data, the data can be
+    intercepted and manipulated. To thwart the manipulation of the data during
+    transmission secure protocol, cipher and algorithms must be used.
+
+severity: medium
+
+identifiers:
+    cce@ocp4: CCE-86231-8
+
+references:
+    nist: CM-6
+    srg: SRG-APP-000014-CTR-000040
+
+ocil_clause: 'insecure TLS version configured in the Custom tlsSecurityProfile used for accessing external services'
+
+ocil: |-
+    To check for the configured tlsSecurityProfile use below command:
+    <tt>$ oc get apiservers.config.openshift.io cluster -o jsonpath='{.spec.tlsSecurityProfile.custom.minTLSVersion}{"\n"}'</tt>
+    The output shows the minTLSVersion configured in Custom tlsSecurityProfile for API Server.
```

Comment:
I think the OCIL should explicitly say that "make sure the tlsProfile is not Old an if it's Custom, the minimum TLS version needs to be 1.2"

Reply:
@bharath-b-rh this still needs fixing (e.g. say explicitly what is permitted and what is not)

Reply:
comment incorporated.